lib/io/can.cpp

```diff
@@ -94,4 +96,24 @@ std::optional<CanFrame> Can::receive()
   return received_message;
 }
 
+void Can::listen()
+{
+  std::optional<CanFrame> data;
+  if (ioctl(socket_, FIONREAD) >= sizeof(CanFrame)) { data = receive(); }
+  if (data == std::nullopt) { return; }
+  const CanFrame message = data.value();
+  if (!processors_.contains(message.can_id)) {
+    logger_.log(core::LogLevel::kInfo, "No CanProccessor associated with id %i", message.can_id);
+  }
+  for (uint8_t i = 0; i < processors_[message.can_id].size(); i++) {
```

Comment:
Lots of things can go wrong here.

Firstly, you need to check whether the key `message.can_id` exists. Then you need to retrieve the value and then you can iterate over it like so:

```cpp
for (auto &processors : subscribed_processors) {
  processor->processMessage(message);
}
```